filament/include/filament/View.h

```diff
@@ -571,6 +572,104 @@ class UTILS_PUBLIC View : public FilamentAPI {
     //! debugging: returns a Camera from the point of view of *the* dominant directional light used for shadowing.
     Camera const* getDirectionalLightCamera() const noexcept;
 
+
+    /** Result of a picking query */
+    struct PickingQueryResult {
+        struct {
+            utils::Entity renderable{};     //! RenderableManager Entity at the queried coordinates
+            uint32_t reserved0{};
+            uint32_t reserved1{};
+            uint32_t reserved2{};
+        };
+    };
+
+    /** User data for PickingQueryResultCallback */
+    struct PickingQuery {
+        void* storage[8];
+    };
+
+    /** callback type used for picking queries. */
+    using PickingQueryResultCallback = void(*)(PickingQueryResult const& result, PickingQuery* pq);
+
+    /**
+     * Helper for creating a picking query from Foo::method, by pointer.
+     * e.g.: auto* query = pick<Foo, &Foo::bar>(x, y, &foo);
+     * @tparam T        Class of the method to call (e.g.: Foo)
+     * @tparam method   Method to call on T (e.g.: &Foo::bar)
+     * @param x         Horizontal coordinate to query in the viewport with origin on the left.
+     * @param y         Vertical coordinate to query on the viewport with origin at the bottom.
+     * @param data      A pointer to an instance of T
+     */
+    template<typename T, void(T::*method)(PickingQueryResult const&)>
+    void pick(uint32_t x, uint32_t y, T* instance) noexcept {
+        PickingQuery& query = pick(x, y, [](PickingQueryResult const& result, PickingQuery* pq) {
+            void* user = pq->storage;
+            (*static_cast<T**>(user)->*method)(result);
+        });
+        query.storage[0] = instance;
+    }
+
+    /**
+     * Helper for creating a picking query from Foo::method, by copy for a small object
+     * e.g.: auto* query = pick<Foo, &Foo::bar>(x, y, foo);
```

Comment:
The function returns void, I don’t understand the example

Reply:
Ah yes, good point, the example is out of daten